research_fish_analysis.py

```diff
@@ -1,288 +1,288 @@
-    dataframeurl = pd.DataFrame({'rootdomains': list_of_rootdomains})
-
-    logger.info('This many rootdomains found: ' + str(len(dataframeurl)))
-
-    logger.info('This many unique rootdomains found: ' + str(len(dataframeurl.rootdomains.unique())))
-
-    return dataframeurl
-
-
-def plot_bar_charts(dataframe,filename,title,xaxis,yaxis,truncate):
-    """
-    Takes a two-column dataframe and plots it
-    :params: a dataframe with two columns (one labels, the other a count), a filename for the resulting chart, a title, and titles for the
-    two axes (if title is None, then nothing is plotted), and a truncate variable which cuts down the number of
-    rows plotted (unless it's 0 at which point all rows are plotted)
-    :return: Nothing, just prints a chart
-    """
-    # Set up logging
-    logger = logging.getLogger(__name__)
-    logger.info('Plotting charts...')
-    
-    if truncate > 0:
-        # This cuts the dataframe down to the number of rows given in truncate
-        dataframe = dataframe.ix[:truncate]
-
-    dataframe.plot(kind='bar', legend=None)
-    plt.title(title)
-    if xaxis != None:
-        plt.xlabel(xaxis)
-    if yaxis != None:
-        plt.ylabel(yaxis)
-    # This provides more space around the chart to make it prettier        
-    plt.tight_layout(True)
-    plt.savefig("./charts/" + filename + '.png', format = 'png', dpi = 150)
-    plt.show()
-    return
-
-
-def impact_to_txt(dataframe,colname):
-    # Set up logging
-    logger = logging.getLogger(__name__)
-    logger.info('Creating impact text file...')
-    
-    # Don't want any of the NaNs, so drop them
-    impact_dataframe = dataframe.dropna(subset=[colname])
-    # Open file for writing
-    file_for_impacts = open("./data/impact.txt", 'w')
-    # Go through dataframe row by row and write the text from the colname column to as a separate line to the text file
-    for i, row in impact_dataframe.iterrows():
-        file_for_impacts.write("%s\n" % impact_dataframe[colname][i])
-    return
-    
-    
-def check_url_status(dataframe, colname, statuscol):
-    logger = logging.getLogger(__name__)
-    logger.info('Checking URLs...')
-    
-    # Don't want any of the NaNs, so drop the rows in which NaN was entered for the URL
-    dataframe.dropna(subset=[colname], inplace=True)
-
-    h = httplib2.Http()
-    for i, row in dataframe.iterrows():    
-        try:
-            print("Checking " + dataframe[colname][i])
-            response, content = h.request(dataframe[colname][i])
-            if response.status < 400:
-                dataframe[statuscol][i] = response['date']
+#!/usr/bin/env python 
+# encoding: utf-8 
+ 
+import pandas as pd 
+from pandas import ExcelWriter 
+import numpy as np 
+import csv 
+import matplotlib.pyplot as plt 
+from urllib.parse import urlparse 
+from collections import Counter 
+import math 
+import requests 
+import httplib2 
+import logging 
+ 
+ 
+def import_xls_to_df(filename, name_of_sheet): 
```

Comment:
👍 

Even though the actual implementation here is just a single line (the call to `pd.read_excel()`), it's nice to have a helper function with a meaningful name for this. Makes the `main()` function below more easily readable.